### PR TITLE
[radio-spinel] fix `GetNextTid()`

### PR DESCRIPTION
**Pull request by abtink**
This commit updates `GetNextTid()` in `RadioSpinel` so that it
searches for the next unused TID. The current implementation only
checked the `mCmdNextTid` and assumed that if it was in-use, then all
other TIDs were also in-use. This may not be valid since a TID may be
in-use for a pending radio tx in `mTxRadioTid` while TIDs after that
are still available.

This should help address the situation where while waiting for a
pending radio tx (holding on to `mTxRadioTid`) 15 other radio API
calls happen (each consuming a TID) causing the `mCmdNextTid` to wrap
around back to `mTxRadioTid`.

Thanks to @dismirlian and @tlifschitz for finding the issue and root-causing it. 👍 

This is related to and should help address issue https://github.com/openthread/openthread/issues/7513.

### CODE REVIEW

**Review by bukepo — APPROVED** [review]
LGTM:+1:
Thanks for fixing this issue!

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

File: src/lib/spinel/radio_spinel_impl.hpp [hunk 1]
```diff
@@ -1699,15 +1699,25 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::WaitResponse(void)
 template <typename InterfaceType, typename ProcessContextType>
 spinel_tid_t RadioSpinel<InterfaceType, ProcessContextType>::GetNextTid(void)
 {
-    spinel_tid_t tid = 0;
+    spinel_tid_t tid = mCmdNextTid;
 
-    if (((1 << mCmdNextTid) & mCmdTidsInUse) == 0)
+    while (((1 << tid) & mCmdTidsInUse) != 0)
```
Comment (simonlingoogle) [inline src/lib/spinel/radio_spinel_impl.hpp:1704]: `mCmdTidsInUse` is `uint16_t`, so the valid TID range should be [1~15].

Does the issue mean that we should increase the range of TID to [1~31] ?

@abtink @bukepo

Comment (abtink) [reply src/lib/spinel/radio_spinel_impl.hpp:1704]: TID is 4 bit value in spinel protocol. So its range [0-15]
https://github.com/openthread/openthread/blob/119df62cbddce9d3e10b0a56d7fbf9d53c7fa218/src/lib/spinel/spinel.h#L59-L85

> Does the issue mean that we should increase the range of TID to [1~31] ?

No. We don't need do. 

`RadioSpinel` seems to only use at most two TIDs at a time (one for radio tx frame, one for a command for which it is waiting for response). The issue here was that `GetNextTid()` only checked one value `mCmdNxtTid` and not the rest of TID values.  More details on this in #7513.

Comment (simonlingoogle) [reply src/lib/spinel/radio_spinel_impl.hpp:1704]: I got what you mean. Thanks for clarification.

**Review by jwhui — APPROVED** [review]

**Pull request merged**